File: Baekjoon/Problem_1000~1999/Problem_1339/main.cpp
#include <cstdio>
#include <iostream>
#include <string>
#include <algorithm>

using namespace std;
const int SIZE = 26;

int N;
int char2int[SIZE];

struct compare{
    bool operator()(const int& a, const int& b) {
        return a > b;
    }
};

int main(void) {
    cin >> N;

    for (int i = 0; i < N; i++) {
        string s; cin >> s;

        for (int i = 0; i < s.size(); i++) {
            char &c = s[i];
            char2int[c - 'A'] += pow(10, s.size()-1-i);
        }
    }

    sort(char2int, char2int + SIZE, compare());
    
    int _sum = 0;
    for (int i = 0; i < 10; i++) {
        _sum += char2int[i] * (9 - i);
    }
    printf("%d\n", _sum);
    
    return 0;
}

File: Baekjoon/Problem_1000~1999/Problem_1339/main.py
import sys

input = lambda: sys.stdin.readline().rstrip()

if __name__ == "__main__":
    N = int(input())
    
    char2int = [ 0 ] * 26
    
    for i in range(N):
        s = input()
        for i in range(len(s)):
            c = s[i]
            char2int[ord(c) - ord('A')] += 10 ** (len(s)-1-i)
    
    char2int.sort(reverse=True)
    
    _sum = 0
    for i in range(10):
        _sum += char2int[i] * (9 - i)
        
    print(_sum)